src/main/java/frc/robot/subsystems/ArmControlSubsystem.java

```diff
@@ -0,0 +1,306 @@
+import edu.wpi.first.wpilibj.DutyCycleEncoder;
+import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.FunctionalCommand;
+import edu.wpi.first.wpilibj2.command.PrintCommand;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.constants.ArmConstants;
+import frc.robot.constants.CompConstants;
+
+
+public class ArmControlSubsystem extends SubsystemBase {
+
+  public enum ArmMotorMode {
+    COAST,
+    BRAKE,
+    OFF
+  }
+
+  public enum MoveArmConfig {
+    SIMULTANEOUS,
+    SEQ_EXTEND_THEN_PIVOT,
+    SEQ_PIVOT_THEN_EXTEND
+  }
+  
+  
+  private final WPI_TalonFX _leftPivotController = new WPI_TalonFX(ArmConstants.kleftArmPivotID);
+  private final WPI_TalonFX _rightPivotController = new WPI_TalonFX(ArmConstants.kRightArmPivotID);
+  private final CANSparkMax extensionController = new CANSparkMax(ArmConstants.kTelescopicArmSparkID, MotorType.kBrushless);
+
+  private final DutyCycleEncoder _absPivEncoder = new DutyCycleEncoder(ArmConstants.kDIOPortPiv);
+  private final RelativeEncoder extensionEncoder = extensionController.getEncoder();
+
+  private final SlewRateLimiter _pivotRateLimiter;
+  private final PIDController _extensionPID;
+  private final PIDController _pivotPID;
+
+  // This is called Ultra Instinct because setting this boolean to true removes the clamps
+  // so that if the belt skips, the operator can still run manually without the bad offsets
+  public static boolean _ultraInstinct = false;
+
+  private double _pivotRelEncoderOffsetRot;
+  private double _currentPivotRotation;
+  private double _desiredPivotRotation = ArmConstants.kMinAngleRad;
+
+  private double _currentExtensionDistance = ArmConstants.kZeroExtensionIn;
+  private double _desiredExtensionDistance = ArmConstants.kMinExtensionIn;
+
+
+
+  private boolean _isInitialized = false;
+
+  public SendableChooser<ArmMotorMode> chooser = new SendableChooser<>();
+
+
+  public ArmControlSubsystem() {
+
+    _pivotPID = new PIDController(1.4, 0, 0);
+    _pivotPID.setTolerance(Units.degreesToRadians(0));
+
+    _extensionPID = new PIDController(0.19, 0, 0);
+    _extensionPID.setTolerance(.2);
+
+    _pivotRateLimiter = new SlewRateLimiter(ArmConstants.kMaxPivotRateRadSec);
+
+    _absPivEncoder.setConnectedFrequencyThreshold(975); //do not change this number pls or else 
+   
+    this._pivotRelEncoderOffsetRot = -_absPivEncoder.getAbsolutePosition() * ArmConstants.kPivotAbsEncToRotation + ArmConstants.kPivotInitOffsetRot;
+   
+    setConfig();
+
+    SmartDashboard.putNumber("kG", ArmConstants.kG);
+    
+    chooser.addOption("Brake", ArmMotorMode.BRAKE);
+    chooser.addOption("Coast", ArmMotorMode.COAST); 
+    chooser.setDefaultOption("Brake", ArmMotorMode.BRAKE);
+
+    SmartDashboard.putData("ArmMotorMode", chooser);
+    
+  }
+
+  public void setConfig(){
```

Comment:
This seems to be initializing to some default state, right? Maybe call this setDefaultConfig?

Reply:
done